File: app/src/main/java/com/koktajlbar/incognitobook/viewmodels/CocktailRowVM.java
package com.koktajlbar.incognitobook.viewmodels;

import android.content.Context;
import android.content.Intent;
import android.view.View;

import androidx.databinding.BaseObservable;

import com.koktajlbar.incognitobook.model.Cocktail;
import com.koktajlbar.incognitobook.view.CocktailDetailsActivity;


public class CocktailRowVM extends BaseObservable {

    private Cocktail cocktail;
    private Context context;

    public CocktailRowVM(Cocktail cocktail, Context context) {
        this.cocktail = cocktail;
        this.context = context;
    }

    public void setCocktail(Cocktail cocktail) {
        this.cocktail = cocktail;
        notifyChange();
    }

    public String getName(){ return cocktail.getName();}

    public void onItemClick(View view) {
        Intent intent = new Intent(context, CocktailDetailsActivity.class);
        intent.putExtra("uuid", cocktail.getId());
        context.startActivity(intent);
    }

    public boolean onLongClick(View view)
    {
        return true;
    }
}

File: app/src/main/java/com/koktajlbar/incognitobook/viewmodels/CocktailRowVM.kt
package com.koktajlbar.incognitobook.viewmodels

import android.content.Context
import android.content.Intent
import android.view.View
import androidx.databinding.BaseObservable
import com.koktajlbar.incognitobook.model.Cocktail
import com.koktajlbar.incognitobook.view.CocktailDetailsActivity

class CocktailRowVM(private var cocktail: Cocktail) : BaseObservable() {
    fun setCocktail(cocktail: Cocktail) {
        this.cocktail = cocktail
        notifyChange()
    }

    val name: String
        get() = cocktail.name

    fun onItemClick(view: View?) {
        val context = view?.context
        val intent = Intent(context, CocktailDetailsActivity::class.java)
        intent.putExtra("uuid", cocktail.id)
        context!!.startActivity(intent)
    }

    fun onLongClick(view: View?): Boolean {
        return true
    }

}